tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,218 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/webclient"
+	autoupdatev1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1"
+	"github.com/gravitational/teleport/api/types/autoupdate"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+// versionResponse is structure for formatting the autoupdate version response.
+type versionResponse struct {
+	TargetVersion string `json:"target_version"`
+}
+
+// AutoUpdateCommand implements the `tctl autoupdate` command for managing
+// autoupdate process for tools and agents.
+type AutoUpdateCommand struct {
+	app    *kingpin.Application
+	config *servicecfg.Config
+
+	upsertCmd *kingpin.CmdClause
+	getCmd    *kingpin.CmdClause
+	watchCmd  *kingpin.CmdClause
+
+	mode               string
+	toolsTargetVersion string
+	proxy              string
+
+	// stdout allows to switch standard output source for resource command. Used in tests.
+	stdout io.Writer
+}
+
+// Initialize allows AutoUpdateCommand to plug itself into the CLI parser.
+func (c *AutoUpdateCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	c.app = app
+	c.config = config
+	autoUpdateCmd := app.Command("autoupdate", "Teleport auto update commands.")
+
+	clientToolsCmd := autoUpdateCmd.Command("client-tools", "Client tools auto update commands.")
+
+	c.upsertCmd = clientToolsCmd.Command("update", "Edit client tools auto update configuration.")
```

Comment:
For the user experience it looks like the update, when resource wasn't created from api we return empty value. Originally in RFD it was defined as `update`, if you think that `upsert` looks better for understanding I can change it here and in RFD

Reply:
I think `update` here will very likely understood as "update the clients" rather than "update the AU configuration resource". The help message is `"Edit client tools auto update configuration."`, so maybe `configure` would be a better fit.